From 9e2227f80fb21ce8e1340214826af50815e292b7 Mon Sep 17 00:00:00 2001
From: "Lu.Jiang" <lu.jiang@windriver.com>
Date: Thu, 4 Sep 2014 19:11:13 +0800
Subject: [PATCH 114/987] b4qds: DPAA support in device trees of B4860 and
 B4420

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

B4860 has 6 1G SGMII and 2 10G interfaces
B4420 has 4 1G SGMII and no 10G

The QDS board has 2 on board SGMII PHYs and other 1G interfaces can be
exercised using SGMII Riser card mounted on a slot on AMC card.
10G interfaces can be used using XAUI riser cards on slots on AMC card.

Also included e6500_power_isa.dtsi in b4si-post.dtsi to match up with T4.

Signed-off-by: Sandeep Singh <sandeep@freescale.com>
Signed-off-by: Suresh Gupta  <suresh.gupta@freescale.com>
Signed-off-by: Poonam Aggrwal <poonam.aggrwal@freescale.com>
Change-Id: I58231f43a75ebfa63bb9c9b3f6717848f0f6f6ed
Reviewed-on: http://git.am.freescale.net:8181/1358
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Signed-off-by: Lu.Jiang <lu.jiang@windriver.com>
[Fix context to apply to WRL.]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/powerpc/boot/dts/b4420qds.dts          | 15 ++++++
 arch/powerpc/boot/dts/b4860qds.dts          | 83 +++++++++++++++++++++++++++++
 arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi  | 15 ++++++
 arch/powerpc/boot/dts/fsl/b4860si-post.dtsi | 15 ++++++
 arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi  | 18 ++++++-
 arch/powerpc/boot/dts/fsl/b4si-post.dtsi    | 70 ++++++++++++++++++++++++
 6 files changed, 214 insertions(+), 2 deletions(-)

diff --git a/arch/powerpc/boot/dts/b4420qds.dts b/arch/powerpc/boot/dts/b4420qds.dts
index 508dbdf..5289cc1 100644
--- a/arch/powerpc/boot/dts/b4420qds.dts
+++ b/arch/powerpc/boot/dts/b4420qds.dts
@@ -45,6 +45,21 @@
 		};
 	};
 
+	fsl,dpaa {
+		compatible = "fsl,b4420-dpaa", "fsl,dpaa";
+		ethernet@0 {
+			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@1 {
+			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@2 {
+			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@3 {
+			compatible = "fsl,b4420-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+	};
 };
 
 /include/ "fsl/b4420si-post.dtsi"
diff --git a/arch/powerpc/boot/dts/b4860qds.dts b/arch/powerpc/boot/dts/b4860qds.dts
index 5d0fa4c..af15f44 100644
--- a/arch/powerpc/boot/dts/b4860qds.dts
+++ b/arch/powerpc/boot/dts/b4860qds.dts
@@ -39,12 +39,64 @@
 	model = "fsl,B4860QDS";
 	compatible = "fsl,B4860QDS";
 
+	aliases{
+		phy_sgmii_1e = &phy_sgmii_1e;
+		phy_sgmii_1f = &phy_sgmii_1f;
+
+		phy_xaui_slot1 = &phy_xaui_slot1;
+		phy_xaui_slot2 = &phy_xaui_slot2;
+	};
+
 	ifc: localbus@ffe124000 {
 		board-control@3,0 {
 			compatible = "fsl,b4860qds-fpga", "fsl,fpga-qixis";
 		};
 	};
 
+	soc: soc@ffe000000 {
+		fman0: fman@400000 {
+			fm1mac5: ethernet@e8000 {
+				phy-handle = <&phy_sgmii_1e>;
+				phy-connection-type = "sgmii";
+			};
+			fm1mac6: ethernet@ea000 {
+				phy-handle = <&phy_sgmii_1f>;
+				phy-connection-type = "sgmii";
+			};
+			fm1mac9: ethernet@f0000 { /* FM1@TGEC2 */
+				phy-handle = <&phy_xaui_slot1>;
+				phy-connection-type = "xgmii";
+			};
+			fm1mac10: ethernet@f2000 { /* FM1@TGEC1 */
+				phy-handle = <&phy_xaui_slot2>;
+				phy-connection-type = "xgmii";
+			};
+			mdio0: mdio@fc000 {
+				phy_sgmii_1e: ethernet-phy@1e {
+					status = "disabled";
+					reg = <0x1e>;
+				};
+				phy_sgmii_1f: ethernet-phy@1f {
+					status = "disabled";
+					reg = <0x1f>;
+				};
+			};
+
+			xmdio0: mdio@fd000 {
+				/* For 10g interfaces */
+				phy_xaui_slot1: xaui-phy@slot1 {
+					status = "disabled";
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x7>; /* default switch setting on slot1 of AMC2PEX */
+				};
+				phy_xaui_slot2: xaui-phy@slot2 {
+					status = "disabled";
+					compatible = "ethernet-phy-ieee802.3-c45";
+					reg = <0x6>; /* default switch setting on slot1 of AMC2PEX */
+				};
+			};
+		};
+	};
 
 	rio: rapidio@ffe0c0000 {
 		reg = <0xf 0xfe0c0000 0 0x11000>;
@@ -56,6 +108,37 @@
 			ranges = <0 0 0xc 0x30000000 0 0x10000000>;
 		};
 	};
+	fsl,dpaa {
+		compatible = "fsl,b4860-dpaa", "fsl,dpaa";
+		ethernet@0 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@1 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@2 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@3 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+		};
+		ethernet@4 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac5>;
+		};
+		ethernet@5 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac6>;
+		};
+		ethernet@8 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac9>;
+		};
+		ethernet@9 {
+			compatible = "fsl,b4860-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&fm1mac10>;
+		};
+	};
 };
 
 /include/ "fsl/b4860si-post.dtsi"
diff --git a/arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi b/arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi
index 338af7e..f2cf23f 100644
--- a/arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4420si-pre.dtsi
@@ -50,6 +50,16 @@
 		serial1 = &serial1;
 		serial2 = &serial2;
 		serial3 = &serial3;
+
+		crypto = &crypto;
+		qman = &qman;
+		bman = &bman;
+		fman0 = &fman0;
+		ethernet0 = &fm1mac1;
+		ethernet1 = &fm1mac2;
+		ethernet2 = &fm1mac3;
+		ethernet3 = &fm1mac4;
+
 		pci0 = &pci0;
 		dma0 = &dma0;
 		dma1 = &dma1;
@@ -61,6 +71,11 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 
+		/*
+		 * Temporarily add next-level-cache info in each cpu node so
+		 * that uboot can do L2 cache fixup. This can be removed once
+		 * u-boot can create cpu node with cache info.
+		 */
 		cpu0: PowerPC,e6500@0 {
 			device_type = "cpu";
 			reg = <0 1>;
diff --git a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
index f2204e3..e93a632 100644
--- a/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4860si-post.dtsi
@@ -178,6 +178,21 @@
 		compatible = "fsl,b4860-l3-cache-controller", "cache";
 	};
 
+/include/ "qoriq-fman3-0-1g-4.dtsi"
+/include/ "qoriq-fman3-0-1g-5.dtsi"
+/include/ "qoriq-fman3-0-10g-0.dtsi"
+/include/ "qoriq-fman3-0-10g-1.dtsi"
+	fman0: fman@400000 {
+		/* tx - 1g - 4 */
+		port@ac000 {
+			fsl,qman-channel-id = <0x806>;
+		};
+		/* tx - 1g - 5 */
+		port@ad000 {
+			fsl,qman-channel-id = <0x807>;
+		};
+	};
+
 	guts: global-utilities@e0000 {
 		compatible = "fsl,b4860-device-config", "fsl,qoriq-device-config-2.0";
 	};
diff --git a/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi b/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
index 02c957d..d8adc2c 100644
--- a/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4860si-pre.dtsi
@@ -51,20 +51,34 @@
 		serial2 = &serial2;
 		serial3 = &serial3;
 
+		crypto = &crypto;
 		qman = &qman;
 		bman = &bman;
+		fman0 = &fman0;
+		ethernet0 = &fm1mac1;
+		ethernet1 = &fm1mac2;
+		ethernet2 = &fm1mac3;
+		ethernet3 = &fm1mac4;
+		ethernet4 = &fm1mac5;
+		ethernet5 = &fm1mac6;
+		ethernet8 = &fm1mac9;
+		ethernet9 = &fm1mac10;
+
 		pci0 = &pci0;
 		dma0 = &dma0;
 		dma1 = &dma1;
 		sdhc = &sdhc;
-	
 	};
 
-
 	cpus {
 		#address-cells = <1>;
 		#size-cells = <0>;
 
+		/*
+		 * Temporarily add next-level-cache info in each cpu node so
+		 * that uboot can do L2 cache fixup. This can be removed once
+		 * u-boot can create cpu node with cache info.
+		 */
 		cpu0: PowerPC,e6500@0 {
 			device_type = "cpu";
 			reg = <0 1>;
diff --git a/arch/powerpc/boot/dts/fsl/b4si-post.dtsi b/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
index 73136c0..2b9747b 100644
--- a/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/b4si-post.dtsi
@@ -287,8 +287,78 @@
 		};
 	};
 
+/include/ "qoriq-qman1.dtsi"
+
+	qman: qman@318000 {
+		interrupts = <16 2 1 28>;
+	};
+
+/include/ "qoriq-bman1.dtsi"
+
+	bman: bman@31a000 {
+		interrupts = <16 2 1 29>;
+	};
+
+/include/ "qoriq-rman-0.dtsi"
+	rman: rman@1e0000 {
+		fsl,qman-channels-id = <0x820 0x821>;
+	};
+
 /include/ "qoriq-mpic4.3.dtsi"
 
+/include/ "qoriq-fman3-0.dtsi"
+/include/ "qoriq-fman3-0-1g-0.dtsi"
+/include/ "qoriq-fman3-0-1g-1.dtsi"
+/include/ "qoriq-fman3-0-1g-2.dtsi"
+/include/ "qoriq-fman3-0-1g-3.dtsi"
+	fman0: fman@400000 {
+		interrupts = <
+			96 2 0 0
+			16 2 1 30>;
+
+		/* tx - 1g - 0 */
+		port@a8000 {
+			fsl,qman-channel-id = <0x802>;
+		};
+		/* tx - 1g - 1 */
+		port@a9000 {
+			fsl,qman-channel-id = <0x803>;
+		};
+		/* tx - 1g - 2 */
+		port@aa000 {
+			fsl,qman-channel-id = <0x804>;
+		};
+		/* tx - 1g - 3 */
+		port@ab000 {
+			fsl,qman-channel-id = <0x805>;
+		};
+		/* offline - 0 is not usable on B4860 */
+		/* offline - 1 */
+		port@82000 {
+			fsl,qman-channel-id = <0x809>;
+		};
+		/* offline - 2 */
+		port@83000 {
+			fsl,qman-channel-id = <0x80a>;
+		};
+		/* offline - 3 */
+		port@84000 {
+			fsl,qman-channel-id = <0x80b>;
+		};
+		/* offline - 4 */
+		port@85000 {
+			fsl,qman-channel-id = <0x80c>;
+		};
+		/* offline - 5 */
+		port@86000 {
+			fsl,qman-channel-id = <0x80d>;
+		};
+		/* offline - 6 */
+		port@87000 {
+			fsl,qman-channel-id = <0x80e>;
+		};
+	};
+
 	guts: global-utilities@e0000 {
 		compatible = "fsl,b4-device-config";
 		reg = <0xe0000 0xe00>;
-- 
1.9.1

